### PR TITLE
PLT-3408 Add SiteURL to config.json

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
Surprisingly, it took very little work to do this on the server. Serverside code can access the site URL either from the Context or directly from the config. Clientside code can get it from mm_config or it can use the TeamStore's methods for convenience. I removed most of the references to window.location along with all of the utility functions in utils.jsx that used it.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3408

#### Checklist
- Includes text changes and localization files updated
- Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

File: model/config.go [hunk 1]
```diff
@@ -358,6 +360,11 @@ func (o *Config) SetDefaults() {
 		o.EmailSettings.PasswordResetSalt = NewRandomString(32)
 	}
 
+	if o.ServiceSettings.SiteURL == nil {
+		o.ServiceSettings.SiteURL = new(string)
+		*o.ServiceSettings.SiteURL = ""
```
Comment (crspeller) [inline model/config.go]: Shouldn't the default be "localhost:8065"?

**Comment by crspeller** [comment]
+1

**Comment by jwilander** [comment]
+1

**Comment by jwilander** [comment]
+1 but we need to update the spinmints to set the correct SiteURL in their config or to not use one since this will be annoying
screen shot 2016-07-31 at 9 37 53 am

**Comment by crspeller** [comment]
Setting to WIP until spinmints are updated.

**Pull request merged**